<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Приложение-шифровальщик</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="topLayout">
      <item>
       <widget class="QLabel" name="cipherLabel">
        <property name="text">
         <string>Выберите шифр:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cipherSelector">
        <item>
         <property name="text">
          <string>Caesar</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Atbash</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Beaufort</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Kuznechik</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>RSA</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>AES-256</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Blowfish</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>3DES</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>CAST5</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="alphabetLabel">
        <property name="text">
         <string>Алфавит:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="alphabetSelector">
        <item>
         <property name="text">
          <string>English</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Russian</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Custom...</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="alphabetDisplayLabel">
      <property name="text">
       <string>Текущий алфавит:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="alphabetDisplay">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QStackedWidget" name="cipherInputStack">
      <!-- Caesar Input Panel -->
      <widget class="QWidget" name="caesarPanel">
       <layout class="QVBoxLayout" name="caesarLayout">
        <item>
         <widget class="QLabel" name="caesarKeyLabel">
          <property name="text">
           <string>Введите сдвиг (число):</string>
          </property>
          <property name="toolTip">
           <string>Укажите числовое значение для сдвига (например, 3).</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="caesarKeyInput">
          <property name="placeholderText">
           <string>Например, 3</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <!-- Atbash Input Panel -->
      <widget class="QWidget" name="atbashPanel">
       <layout class="QVBoxLayout" name="atbashLayout">
        <item>
         <widget class="QLabel" name="atbashInfoLabel">
          <property name="text">
           <string>Шифр Atbash не требует ключа.</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <!-- Beaufort Input Panel -->
      <widget class="QWidget" name="beaufortPanel">
       <layout class="QVBoxLayout" name="beaufortLayout">
        <item>
         <widget class="QLabel" name="beaufortKeyLabel">
          <property name="text">
           <string>Введите ключ (текст):</string>
          </property>
          <property name="toolTip">
           <string>Укажите текстовый ключ для шифра Beaufort.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="beaufortKeyInput">
          <property name="placeholderText">
           <string>Например, KEY</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <!-- Kuznechik Input Panel -->
      <widget class="QWidget" name="kuznechikPanel">
       <layout class="QVBoxLayout" name="kuznechikLayout">
        <item>
         <widget class="QLabel" name="kuznechikKeyLabel">
          <property name="text">
           <string>Ключ (32 байта в hex):</string>
          </property>
          <property name="toolTip">
           <string>Введите 32 байта в шестнадцатеричном формате (например, 8899aabbccddeeff00112233445566778899aabbccddeeff0011223344556677).</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="kuznechikKeyInput">
          <property name="placeholderText">
           <string>Например, 8899aabbccddeeff00112233445566778899aabbccddeeff0011223344556677</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="kuznechikVectorLabel">
          <property name="text">
           <string>Вектор (8 байт в hex):</string>
          </property>
          <property name="toolTip">
           <string>Введите 8 байт в шестнадцатеричном формате (например, 1234567890abcdef).</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="kuznechikVectorInput">
          <property name="placeholderText">
           <string>Например, 1234567890abcdef</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <!-- RSA Input Panel -->
      <widget class="QWidget" name="rsaPanel">
       <layout class="QVBoxLayout" name="rsaLayout">
        <item>
         <widget class="QPushButton" name="generateKeysButton">
          <property name="text">
           <string>Сгенерировать ключи</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="publicKeyLabel">
          <property name="text">
           <string>Публичный ключ:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="publicKeyInput">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string>Публичный ключ появится после генерации</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="privateKeyLabel">
          <property name="text">
           <string>Приватный ключ (для расшифровки):</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="privateKeyInput">
          <property name="placeholderText">
           <string>Введите приватный ключ для расшифровки</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <!-- AES-256 Input Panel -->
      <widget class="QWidget" name="aes256Panel">
       <layout class="QVBoxLayout" name="aes256Layout">
        <item>
         <widget class="QLabel" name="aes256KeyLabel">
          <property name="text">
           <string>Введите ключ (текст):</string>
          </property>
          <property name="toolTip">
           <string>Ключ должен быть не менее 32 символов для AES-256.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="aes256KeyInput">
          <property name="placeholderText">
           <string>Например, 32 символа</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <!-- Blowfish Input Panel -->
      <widget class="QWidget" name="blowfishPanel">
       <layout class="QVBoxLayout" name="blowfishLayout">
        <item>
         <widget class="QLabel" name="blowfishKeyLabel">
          <property name="text">
           <string>Введите ключ (текст):</string>
          </property>
          <property name="toolTip">
           <string>Ключ для Blowfish может быть от 4 до 56 байт.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="blowfishKeyInput">
          <property name="placeholderText">
           <string>Например, mysecretkey</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <!-- 3DES Input Panel -->
      <widget class="QWidget" name="tripleDesPanel">
       <layout class="QVBoxLayout" name="tripleDesLayout">
        <item>
         <widget class="QLabel" name="tripleDesKeyLabel">
          <property name="text">
           <string>Введите ключ (текст):</string>
          </property>
          <property name="toolTip">
           <string>Ключ для 3DES должен быть 24 байта (например, 24 символа).</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="tripleDesKeyInput">
          <property name="placeholderText">
           <string>Например, 24 символа</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <!-- CAST5 Input Panel -->
      <widget class="QWidget" name="cast5Panel">
       <layout class="QVBoxLayout" name="cast5Layout">
        <item>
         <widget class="QLabel" name="cast5KeyLabel">
          <property name="text">
           <string>Введите ключ (текст):</string>
          </property>
          <property name="toolTip">
           <string>Ключ для CAST5 должен быть от 5 до 16 байт.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="cast5KeyInput">
          <property name="placeholderText">
           <string>Например, mycast5key</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="inputTextLabel">
      <property name="text">
       <string>Введите текст:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="inputText"/>
    </item>
    <item>
     <widget class="QLabel" name="outputTextLabel">
      <property name="text">
       <string>Результат:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="outputText">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="buttonLayout">
      <item>
       <widget class="QPushButton" name="encryptButton">
        <property name="text">
         <string>Зашифровать</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="decryptButton">
        <property name="text">
         <string>Расшифровать</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="exportButton">
        <property name="text">
         <string>Экспорт</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
</ui>
